### PR TITLE
Fix | Update node to version 20

### PR DESCRIPTION
**Pull request by cmm-apli**
## Description

Update node to version  20.
This will fix docker error for deployments.
https://github.com/chatwoot/chatwoot/pull/7759

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Locally ran the solution.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by joseangelbadillo12 — APPROVED** [review]

File: .circleci/config.yml
Comment (AranGarcia) [inline .circleci/config.yml:1]: Where are we using Circle CI? 😲

File: .github/workflows/run_foss_spec.yml [hunk 1]
```diff
@@ -74,6 +74,8 @@ jobs:
     - name: Run backend tests
       run: |
         bundle exec rspec --profile=10 --format documentation
+      env: 
+        NODE_OPTIONS: --openssl-legacy-provider
```
Comment (AranGarcia) [inline .github/workflows/run_foss_spec.yml:78]: Why is a legacy provider setting required?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # pre-build stage
-FROM ruby:3.2.2-alpine3.16 AS pre-builder
+FROM ruby:3.2.2-alpine3.18 AS pre-builder
```
Comment (AranGarcia) [inline docker/Dockerfile:2]: Also why is Ruby's docker image being updated, too?

Comment (cmm-apli) [reply docker/Dockerfile:2]: Same here, I was just checking which PR fixed the issues, and figured out it was this, so I just made cherrypick of the PR referenced on description
https://github.com/chatwoot/chatwoot/pull/7759

**Pull request merged**